rust-install/src/component/package.rs

```diff
@@ -77,6 +99,68 @@ impl Package for DirectoryPackage {
 	}
 }
 
+// On Unix we need to set up the file permissions correctly so
+// binaries are executable and directories readable.
```

Comment:
Is this strictly necessary? Can't the permissions be set correctly when the package is built? If not, can this be done as part of `copy_file`/`copy_dir`: I'm wary of doing such a blanket permissions change, and I'd also like to avoid modifying the installation source in any way.